### PR TITLE
Dependency upgrades

### PR DESCRIPTION
**Pull request by k88hudson**
Fix some of #2321. This PR:

- upgrades karma and related dependencies (which mostly fixes inline-sourcemaps)
- upgrades from webpack 2.0 to webpack 3.0 (which has no breaking changes except internal ones - see [the release notes](https://medium.com/webpack/webpack-3-official-release-15fd2dd8f07b))
- upgrades sinon and fixes some deprecated .stub calls that were showing warning messages (`sinon.stub(foo, "bar").returns("baz")` instead of `sinon.stub(foo, "bar", () => "baz")`

I've run the automated tests and did some manual testing, everything seems to be working as expected.

### CODE REVIEW

File: system-addon/test/unit/lib/ManualMigration.test.js [hunk 1]
```diff
@@ -97,7 +97,8 @@ describe("ManualMigration", () => {
     });
     it("should set migrationStatus when isMigrationMessageExpired is true", async () => {
       const setStatusStub = sinon.stub(instance, "expireMigration");
-      sinon.stub(instance, "isMigrationMessageExpired", () => new Promise(resolve => { resolve(true); }));
+      sinon.stub(instance, "isMigrationMessageExpired")
+        .returns(new Promise(resolve => { resolve(true); }));
```
Comment (Mardak) [inline system-addon/test/unit/lib/ManualMigration.test.js]: nit: this should be able to just be `.returns(Promise.resolve(true))` ?

**Comment by Mardak** [comment]
I guess do we want to turn off various automation related to test pilot add-on?

**Comment by k88hudson** [comment]
@Mardak 83a2b73 look OK to you?

**Comment by Mardak** [comment]
Looks good! 👍

**Comment by as-pine-proxy** [comment]
[Treeherder: 9d37661bd70e6df5cf247636c95397bf530f2e56](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=9d37661bd70e6df5cf247636c95397bf530f2e56)

**Pull request merged**